course/include/isometry.h

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file Vector.h
+ * @author Gonzalo Cervetti (g.cervetti@creativa77.com)
+ * @brief This Class provides implementation for vector operation.
+ * @version 0.1
+ * @date 2020-02-03
+ *
+ * @copyright Copyright (c) 2020
+ *
+ */
+
+#pragma once
+
+// Standard libraries
+#include <math.h>
+#include <initializer_list>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace ekumen {
+namespace math {
+
+class Vector3 {
+   public:
+    Vector3(const double& x, const double& y, const double& z);
+    Vector3();
+    /**
+     * @brief Normalizes a function./
+     *
+     * @return double normalized value of the vector.
+     */
+    inline double norm() const { return sqrt(pow(x_, 2.0) + pow(y_, 2.0) + pow(z_, 2.0)); };
+    inline double dot(Vector3 v) const { return ((x_ * v.x_) + (y_ * v.y_) + (z_ * v.z_)); };
+    Vector3 cross(Vector3 v) const;
+
+    // Getter for elements of the vector.
+    const double& x() const { return x_; };
+    const double& y() const { return y_; };
+    const double& z() const { return z_; };
+
+    double& x() { return x_; };
+    double& y() { return y_; };
+    double& z() { return z_; };
+
+    // Operators overloading.
+    inline Vector3 operator+(const Vector3& a) const { return Vector3(x_ + a.x_, y_ + a.y_, z_ + a.z_); };
```

Comment:
Usually you overload the `+=` operator and then use it to implement the `+` operator. Usually you need both, so that way they share the implementation.

Reply:
Same for `-`.